### ISSUE TITLE
Convos wont login when using it with a reverse proxy (nginx) with a path

### ISSUE THREAD

**Issue by Adoliin** [convos-chat/convos#674]
# Describe the bug
When setting convos with a reverse proxy (nginx) and trying to visit the specified url it works the first time but uppon refreshing the page it start querying a wrong URL. And on each refresh that address get appended /irc/ (the path i am using on Nginx).
So I think a loop is messing with the fetch function or something.

# To Reproduce

command to launch Convos
```CONVOS_REVERSE_PROXY=1 ./script/convos daemon --listen http://127.0.0.1:3005```

ngix config file `nginx.conf`
```
server {
        listen 80 default_server;
        listen [::]:80 default_server;

        root /var/www/html;

        index index.html index.htm index.nginx-debian.html;

        server_name _;

        location / {
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection 'upgrade';
                proxy_set_header Host $host;
                proxy_cache_bypass $http_upgrade;
        }

        location /irc/ {
                # Pass requests on to the upstream defined above
                proxy_pass http://127.0.0.1:3005/;

                # Instruct Convos to do the right thing regarding
                # HTTP and WebSockets
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection "upgrade";

                # http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size
                client_max_body_size 0;

                # Enable Convos to construct correct URLs by passing on custom headers.
                proxy_set_header Host $host;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                rewrite ^/irc/?(.*)$ /$1 break;
                proxy_set_header X-Request-Base "$scheme://$host/irc";
        }
}
```

console logs when visiting the url: as you can see there are many appended `/irc` which makes me think that something is messing with the fetch function
```
[route] path="/login" ["login"], query={}
Operation.js:184 GET http://192.168.1.121/irc/irc/irc/irc/irc/irc/irc/api/i18n/en.json 404 (Not Found)
value @ Operation.js:184
(anonymous) @ Operation.js:79
Promise.then (async)
value @ Operation.js:74
(anonymous) @ I18N.js:52
l @ runtime.js:63
(anonymous) @ runtime.js:294
(anonymous) @ runtime.js:119
hn @ asyncToGenerator.js:3
a @ asyncToGenerator.js:25
(anonymous) @ asyncToGenerator.js:32
(anonymous) @ asyncToGenerator.js:21
(anonymous) @ I18N.js:56
Rm @ App.svelte:57
tt @ index.mjs:1809
Lm @ App.svelte:158
(anonymous) @ index.js:22
(anonymous) @ index.js:46
Operation.js:94 http://192.168.1.121/irc/irc/irc/irc/irc/irc/irc/api/i18n/en.json FAILED SyntaxError: Unexpected token < in JSON at position 0
(anonymous) @ Operation.js:94
Promise.catch (async)
value @ Operation.js:92
(anonymous) @ I18N.js:52
l @ runtime.js:63
(anonymous) @ runtime.js:294
(anonymous) @ runtime.js:119
hn @ asyncToGenerator.js:3
a @ asyncToGenerator.js:25
(anonymous) @ asyncToGenerator.js:32
(anonymous) @ asyncToGenerator.js:21
(anonymous) @ I18N.js:56
Rm @ App.svelte:57
tt @ index.mjs:1809
Lm @ App.svelte:158
(anonymous) @ index.js:22
(anonymous) @ index.js:46
Socket.js:264 [socket] open Event {isTrusted: true, type: 'open', target: WebSocket, currentTarget: WebSocket, eventPhase: 2, …}
Socket.js:271 [socket] reconnect in false 
```

# Environment
 - OS: Ubuntu 20.04
 - Browser: chrome, firefox
 - Version: latest

**Comment by mehmooda** [comment]
Got a similar problem:

I'm running the convos/convos:stable docker image with CONVOS_REQUEST_BASE="https://my_host/irc"
and a rewrite with irc/(.*) -> /$1

every time I curl https://my_host/irc/api the base_path get appended further and further
etc
{"basePath":"\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/api",
{"basePath":"\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/api",
{"basePath":"\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/api",
{"basePath":"\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/irc\/api",

It seems to be cached so the first client connects correctly and then errors on every other client.

**Comment by jhthorsen** [comment]
@mehmooda: Please share some of your config. I've set up my own nginx dev-server using this sample config: https://convos.chat/doc/reverse-proxy#example-nginx-config.

**Comment by jhthorsen** [comment]
I think this has gone stale.

**Comment by shambleslol** [comment]
I'm having a similar issu; however, I noticed that the `/login` page's "Sign in" button is not respecting the X-Request-Base path. I used the example nginx config word-for-word and modified the upstream/domain to my instance. 

I can successfully browse to the `https://www.domain.com/convos/` which redirects to `https://www.domain.com/convos/login` but when clicking the "Sign in" button, it keeps trying to contact `https://www.domain.com/api/` and NOT `https://www.domain.com/convos/api`

All the other elements of the page load the correct path. 

Checked in both stable & alpha. 

Any suggestions?
